plugins/secrets/vault/opts.go

```diff
@@ -24,3 +28,21 @@ func WithRenewal(d time.Duration) Opts {
 		v.renew = d
 	}
 }
+
+func WithKubernetesAuth() Opts {
```

Comment:
we should make this `WithKubernetesAuth(addr, role, mount string)` and then in the main.go (enterprise version) I will wire everything up and ensure the variables are passed to the function